examples/dev_your_plugin/plugins/your_plugin.go

```diff
@@ -58,9 +60,44 @@ type filter struct {
 func (f *filter) DecodeHeaders(headers api.RequestHeaderMap, endStream bool) api.ResultAction {
 	hdr := http.Header{}
 	hdr.Set("content-type", "text/plain")
+	allHeaders := headers.GetAllHeaders()
+	msg := "Headers:\n"
+	for k, v := range allHeaders {
+		msg += k + ": " + strings.Join(v, ",") + "\n"
+	}
 	return &api.LocalResponse{
 		Code:   200,
-		Msg:    "Your plugin is run\n",
+		Msg:    "Your plugin is running\n" + msg,
 		Header: hdr,
 	}
 }
+
+// DecodeData might be called multiple times during handling the request body.
+// The endStream is true when handling the last piece of the body.
+func (f *filter) DecodeData(data api.BufferInstance, endStream bool) api.ResultAction {
+	if !endStream {
+		return api.WaitAllData
+	}
+	requestBody := data.Bytes()
+	fmt.Println("Request body: ", string(requestBody))
```

Comment:
Better to use api.LogXX methods to generate the message

Reply:
does it really work since I'm not using log inside `onLog`?

Reply:
`onLog` is called when generating access log while `api.LogXX` can be used anywhere to generate application log.

Reply:
envoy itself will generate the accessLog for all request (including those have plugins or not), what are the scenarios that we need to print additional access logs?

Reply:
Sometimes we need to report more info. For example, in the onLog we can set some dynamic metadata according to the execution result of a request, to enrich the content of access log.

Note that in Envoy 1.31, the filters' onLog phase is executed after the access log is generated. This behavior is changed in the main branch. We have modified Envoy to behave like this inside Ant Group.

Reply:
Do we enrich the original log, or just double the access log with more information?

Reply:
Enrich the original one. For example, you can add a field `%DYNAMIC_METADATA(htnn:your_plugin:blahblah)%` in the access log format.